tests/unit/lms/views/api/canvas/sync_test.py

```diff
@@ -99,6 +114,26 @@ def test_get_canvas_groups_instructor(pyramid_request, canvas_api_client):
     canvas_api_client.group_category_groups.assert_called_once_with(group_set)
 
 
+@pytest.mark.usefixtures("user_is_instructor", "is_group_launch")
+def test_get_canvas_groups_instructor_empty(pyramid_request, canvas_api_client):
+    # pylint: disable=protected-access
+    canvas_api_client.group_category_groups.return_value = []
+
+    with pytest.raises(CanvasGroupSetEmpty):
+        Sync(pyramid_request)._get_canvas_groups()
```

Comment:
The testing of the private method probably suggests that `_get_canvas_groups()` belongs in `CanvasAPIClient`